src/Microsoft.AspNetCore.Build/targets/Compile.targets

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <!-- Put targets attached to the 'Compile' phase of the Standard Lifecycle here. -->
+
+    <Target Name="DotNetBuild" BeforeTargets="Compile">
+        <ItemGroup>
+            <_ToBuild Include="@(Projects)" Condition="'$(BuildSourceOnly)' != 'true' Or '%(Projects.ProjectGroup)' == 'src'" />
+        </ItemGroup>
+
+        <Delete Files="%(_ToBuild.GeneratedBuildInfoFile)" Condition="Exists(%(_ToBuild.GeneratedBuildInfoFile))" />
+        <WriteLinesToFile
```

Comment:
It does always do this for now. We can re-examine if we want, but I don't think the impact is high enough to make it conditional right now. I can certainly make it conditional on CommitHash having a value (and then we can move that decision to the code that sets CommitHash)